nflow-explorer/src/app/workflow/workflow.js

```diff
@@ -4,15 +4,29 @@
   var m = angular.module('nflowExplorer.workflow', [
     'nflowExplorer.workflow.graph',
     'nflowExplorer.workflow.info',
-    'nflowExplorer.workflow.tabs'
+    'nflowExplorer.workflow.tabs',
+    'nflowExplorer.services.WorkflowService'
   ]);
 
   m.controller('WorkflowCtrl', function (workflow, definition,
-                                          parentWorkflow, childWorkflows) {
+      parentWorkflow, childWorkflows, $scope, config, $interval, WorkflowService) {
     var self = this;
     self.workflow = workflow;
     self.parentWorkflow = parentWorkflow;
     self.definition = definition;
     self.childWorkflows = childWorkflows;
+
+    function reloadWorkflow(workflowId) {
+      console.log('Fetching workflow id ' + workflowId);
+      WorkflowService.get(workflowId).then(function(workflow) {
```

Comment:
If there is already a pending fetch - should we cancel it or skip this fetch? This is for example when the network is down/slow.

Reply:
Good point, though the previous pollers (executors, workflow definition) don't do this either. Maybe we should add a task for fixing all the pollers?